inst/@sym/jordan.m

```diff
@@ -0,0 +1,170 @@
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @deftypefn  {Function File} {@var{J} =} jordan (@var{A})
+%% @deftypefnx {Function File} {[@var{V}, @var{J}] =} jordan (@var{A})
+%% Symbolic Jordan canonical form of a matrix.
+%%
+%% Example:
+%% @example
+%% @group
+%% A = sym ([6 5 -2 -3; -3 -1 3 3; 2 1 -2 -3; -1 1 5 5]);
+%% jordan (A)
+%%   @result{} ans = (sym 4×4 matrix)
+%%       ⎡2  1  0  0⎤
+%%       ⎢          ⎥
+%%       ⎢0  2  0  0⎥
+%%       ⎢          ⎥
+%%       ⎢0  0  2  1⎥
+%%       ⎢          ⎥
+%%       ⎣0  0  0  2⎦
+%% @end group
+%% @end example
+%%
+%% We can also compute the generalized eigenvectors:
+%% @example
+%% @group
+%% [V, J] = jordan (A)
+%%   @result{} V = (sym 4×4 matrix)
+%%       ⎡12  1  -26  -2⎤
+%%       ⎢              ⎥
+%%       ⎢-6  2  18   -3⎥
+%%       ⎢              ⎥
+%%       ⎢0   1  -10  0 ⎥
+%%       ⎢              ⎥
+%%       ⎣6   0   2   1 ⎦
+%%   @result{} J = (sym 4×4 matrix)
+%%       ⎡2  1  0  0⎤
+%%       ⎢          ⎥
+%%       ⎢0  2  0  0⎥
+%%       ⎢          ⎥
+%%       ⎢0  0  2  1⎥
+%%       ⎢          ⎥
+%%       ⎣0  0  0  2⎦
+%% @end group
+%% @end example
+%% The generalized eigenvectors are the columns of @var{V}.
+%% Those corresponding to a jordan block form a cycle.
+%% We can check those columns corresponding to the leftmost jordan block:
+%% @example
+%% @group
+%% B = A - J(2, 2) * eye (4)
+%%   @result{} B = (sym 4×4 matrix)
+%%       ⎡4   5   -2  -3⎤
+%%       ⎢              ⎥
+%%       ⎢-3  -3  3   3 ⎥
+%%       ⎢              ⎥
+%%       ⎢2   1   -4  -3⎥
+%%       ⎢              ⎥
+%%       ⎣-1  1   5   3 ⎦
+%% v2 = V(:, 2)
+%%   @result{} v2 = (sym 4×1 matrix)
+%%       ⎡1⎤
+%%       ⎢ ⎥
+%%       ⎢2⎥
+%%       ⎢ ⎥
+%%       ⎢1⎥
+%%       ⎢ ⎥
+%%       ⎣0⎦
+%% v1 = B * v2
+%%   @result{} v1 = (sym 4×1 matrix)
+%%       ⎡12⎤
+%%       ⎢  ⎥
+%%       ⎢-6⎥
+%%       ⎢  ⎥
+%%       ⎢0 ⎥
+%%       ⎢  ⎥
+%%       ⎣6 ⎦
+%% v0 = B * v1
+%%   @result{} v0 = (sym 4×1 matrix)
+%%       ⎡0⎤
+%%       ⎢ ⎥
+%%       ⎢0⎥
+%%       ⎢ ⎥
+%%       ⎢0⎥
+%%       ⎢ ⎥
+%%       ⎣0⎦
+%% @end group
+%% @end example
+%%
+%% @seealso{@@sym/charpoly, @@sym/eig}
+%% @end deftypefn
+
+%% Author: Alex Vong
+%% Keywords: symbolic
+
+function [V, J] = jordan (A)
+  cmd = {'(A,) = _ins'
+         'if A.is_Matrix:'
+         '    (V, J) = A.jordan_form ()'
+         'else:'
+         '    (V, J) = sp.Matrix ([A]).jordan_form ()'};
```

Comment:
not a big deal, but I usually follow python conventions within the python blocks: so no weird octave spaces before the `()`.

This is just FYI, up to you if you change it.